src/linear_algebra/determinant.lean

```diff
@@ -319,4 +330,208 @@ begin
+begin
+  unfold det,
+  rw sum_mul_sum,
+  let preserving_A : finset (perm (m ⊕ n)) :=
+    univ.filter (λ σ, ∀ x, ∃ y, sum.inl y = (σ (sum.inl x))),
+  simp_rw univ_product_univ,
+  have mem_preserving_A : ∀ {σ : perm (m ⊕ n)},
+    σ ∈ preserving_A ↔ ∀ x, ∃ y, sum.inl y = σ (sum.inl x) :=
+    λ σ, mem_filter.trans ⟨λ h, h.2, λ h, ⟨mem_univ _, h⟩⟩,
+  rw ← sum_subset (subset_univ preserving_A) _,
+  rw (sum_bij (λ (σ : perm m × perm n) _, equiv.sum_congr σ.fst σ.snd) _ _ _ _).symm,
+  { intros a ha,
+    rw mem_preserving_A,
+    intro x,
+    use a.fst x,
+    simp },
+  { simp only [forall_prop_of_true, prod.forall, mem_univ],
+    intros σ₁ σ₂,
+    rw fintype.prod_sum_type,
+    simp_rw [equiv.sum_congr_apply, sum.map_inr, sum.map_inl, from_blocks_apply₁₁,
+      from_blocks_apply₂₂],
+    have hr : ∀ (a b c d : R), (a * b) * (c * d) = a * c * (b * d), { intros, ring },
+    rw hr,
+    congr,
+    norm_cast,
+    rw sign_sum_congr },
+  { intros σ₁ σ₂ h₁ h₂,
+    dsimp only [],
+    intro h,
+    dunfold equiv.sum_congr at h, simp only [] at h,
+    have h2 : ∀ x, sum.map (σ₁.fst) (σ₁.snd) x = sum.map (σ₂.fst) (σ₂.snd) x :=
+      λ (x : m ⊕ n), congr_fun h.left x,
+    have h3 := sum.forall.mp h2,
+    simp only [sum.map_inr, sum.map_inl] at h3,
+    ext,
+    { exact h3.left x },
+    { exact h3.right x }},
+  { intros σ hσ,
+    have h1 : ∀ (x : m ⊕ n), (∃ (a : m), sum.inl a = x) → (∃ (a : m), sum.inl a = σ x),
+    { rintros x ⟨a, ha⟩,
+      rw ← ha,
+      exact (@mem_preserving_A σ).mp hσ a },
+    have h2 : ∀ (x : m ⊕ n), (∃ (b : n), sum.inr b = x) → (∃ (b : n), sum.inr b = σ x),
+    { rintros x ⟨b, hb⟩,
+      rw ← hb,
+      exact perm_on_inr_of_perm_on_inl σ ((@mem_preserving_A σ).mp hσ) b },
+    let σ₁' := subtype_perm_of_fintype σ h1,
+    let σ₂' := subtype_perm_of_fintype σ h2,
+    let σ₁ := perm_congr (equiv.set.range (@sum.inl m n) sum.injective_inl).symm σ₁',
+    let σ₂ := perm_congr (equiv.set.range (@sum.inr m n) sum.injective_inr).symm σ₂',
+    use [⟨σ₁, σ₂⟩, finset.mem_univ _],
+    ext,
+    cases x with a b,
+    { rw [equiv.sum_congr_apply, sum.map_inl, perm_congr_apply, equiv.symm_symm,
+        set.apply_range_symm (@sum.inl m n)],
+      erw subtype_perm_apply,
+      rw [set.range_apply, subtype.coe_mk, subtype.coe_mk] },
+    { rw [equiv.sum_congr_apply, sum.map_inr, perm_congr_apply, equiv.symm_symm,
+        set.apply_range_symm (@sum.inr m n)],
+      erw subtype_perm_apply,
+      rw [set.range_apply, subtype.coe_mk, subtype.coe_mk] }},
+  { intros σ h0 hσ,
+    obtain ⟨a, ha⟩ := not_forall.mp ((not_congr (@mem_preserving_A σ)).mp hσ),
+    generalize hx : σ (sum.inl a) = x,
+    cases x with a2 b,
+    { have hn := (not_exists.mp ha) a2,
+      exact absurd hx.symm hn },
+    { rw [finset.prod_eq_zero (finset.mem_univ (sum.inl a)), mul_zero],
+      rw [hx, from_blocks_apply₂₁], refl }}
+end
+
+lemma index_equiv_det (f : equiv m n) (N : matrix n n R)
+  : matrix.det (λ i j, N (f i) (f j)) = N.det :=
```

Comment:
This statement is precisely `matrix.det_reindex_alg_equiv`

Reply:
Thanks! Fixed.